plugins/content/attachment/lib/api/KalturaAttachmentAsset.php

```diff
@@ -45,4 +45,21 @@ public function getMapBetweenObjects ( )
 	{
 		return array_merge ( parent::getMapBetweenObjects() , self::$map_between_objects );
 	}
+
+	public function toObject($object_to_fill = null, $props_to_skip = array())
+	{
+		if (!$object_to_fill)
+		{
+			$className = AttachmentPlugin::getObjectClass('asset', AttachmentPlugin::getAssetTypeCoreValue(AttachmentAssetType::ATTACHMENT));
+			$object_to_fill = new $className();
+		}
+
+		return parent::toObject($object_to_fill, $props_to_skip);
+	}
+
+	public function getInstance()
```

Comment:
1. Please implement IApiObjectFactory and use similar signature.
2. If you always return the class of $this, what's the point of this method?
3. Please use it for all actions, get, update, list, etc.